### PR TITLE
Feature/165 use pylint - add to travis.ci

### PR DESCRIPTION
**Pull request by mitzimorris**
#### Submission Checklist

- [x] Run unit tests
- [x] Declare copyright holder and open-source license: see below

#### Summary
- ran pylint on all tests in dir `test`, fixed problems accordingly

- added pylint to travis workflow for dirs `cmdstanpy` and `test`

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company): Columbia University

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)

### CODE REVIEW

**Comment by wesbarnett** [comment]
Should the flake8 line be removed if pylint is being used? 

We also would probbaly to lint the test directory (I think I had that in another PR).

**Comment by mitzimorris** [comment]
> Should the flake8 line be removed if pylint is being used?

does pylint include all the flake8 checks?

> We also would probably want to lint the test directory (I think I had that in another PR with flake8).

agreed.

**Comment by mitzimorris** [comment]
leaving both flake8 and pylint - flake8 has add'l checks on formatting.

**Comment by mitzimorris** [comment]
OK - ready for code review - changes to the config for pylint and travis, cleanup of unit tests, no changes to cmdstanpy modules.

File: test/test_cmdstan_args.py [hunk 1]
```diff
@@ -434,41 +433,21 @@ def test_args_bad(self):
...
-            # bad inits
-            cmdstan_args = CmdStanArgs(
-                model_name='bernoulli',
-                model_exe='bernoulli.exe',
-                chain_ids=[1, 2, 3, 4],
-                inits='no/such/path/to.file',
-                method_args=sampler_args,
-            )
-
-        with self.assertRaises(ValueError):
```
Comment (wesbarnett) [inline test/test_cmdstan_args.py]: Was this accidentally removed? Looks like it was duplicated code and both got removed in this.

Comment (mitzimorris) [reply test/test_cmdstan_args.py]: it was triplicated code!
removed 2 copies, left one.

Comment (mitzimorris) [reply test/test_cmdstan_args.py]: whoa, I take that back - you're correct.

**Pull request merged**